p2panda-discovery/src/mdns/mod.rs

```diff
@@ -39,23 +43,63 @@ pub struct LocalDiscovery {
     tx: Sender<Message>,
 }
 
+/// Create a new network monitor and subscribe to major interface changes.
+async fn network_monitor() -> Result<Receiver<bool>> {
+    let network_monitor = Monitor::new().await?;
+    let (interface_change_tx, interface_change_rx) = mpsc::channel(8);
+    let _token = network_monitor
+        .subscribe(move |is_major| {
+            debug!("detected major network interface change");
+            let interface_change_tx = interface_change_tx.clone();
+            async move {
+                interface_change_tx.send(is_major).await.ok();
+            }
+            .boxed()
+        })
+        .await?;
+
+    Ok(interface_change_rx)
+}
+
+impl Default for LocalDiscovery {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
 impl LocalDiscovery {
-    pub fn new() -> Result<Self> {
+    pub fn new() -> Self {
         let (tx, rx) = flume::bounded(64);
 
-        let socket = socket_v4()?;
+        let mut socket_is_bound = false;
+        let mut socket = match socket_v4() {
+            Ok(socket) => {
+                socket_is_bound = true;
+                socket
+            }
+            Err(err) => {
+                warn!("failed to create udp socket for mdns discovery: {}", err);
+                socket_v4_unbound().expect("create udp socket")
```

Comment:
This is the tradeoff we have for being able to remove the `Result` return type. The "unbound" socket creation involves a low-level system call and doesn't bind to any interface or address. The likelihood of an error here seems very low and would likely represent a critical issue.